library/core/src/slice/memchr.rs

```diff
@@ -38,19 +39,30 @@ fn repeat_byte(b: u8) -> usize {
 }
 
 /// Returns the first index matching the byte `x` in `text`.
+#[inline]
 pub fn memchr(x: u8, text: &[u8]) -> Option<usize> {
+    // Fast path for small slices
+    if text.len() < 2 * USIZE_BYTES {
+        return text.iter().position(|elt| *elt == x);
+    }
+
+    memchr_general_case(x, text)
+}
+
+#[inline(never)]
```

Comment:
couldn't this inhibit const folding by llvm in some cases?

Reply:
At least in my godbolt-example, as far as I can tell, after 16 elements the whole memchr code would be inlined instead of folding. So technically, yes, but I'm not sure in practice.

Reply:
On the other hand, currently there is no const folding here anyway :)